client/src/components/Dashboard.js

```diff
@@ -141,10 +143,30 @@ class Dashboard extends Component {
     });
   };
 
+  handleToggle = languages => {
+    this.setState({ toggled: !this.state.toggled }, () => {
+      if (!this.state.toggled) {
+        this.setDocuments();
+      } else {
+        const filtereddDocs = this.state.documents.filter(document => {
+          return (
+            languages.includes(document.from_language_name) &&
+            languages.includes(document.to_language_name)
+          );
+        });
+        this.setState({ documents: filtereddDocs });
+      }
+    });
```

Comment:
This works but there are a couple of things you can improve here. Currently, if the toggle is off, you call `setDocuments` again to reload all the documents again because your `documents` state doesn't contain everything after you filter. This may be inefficient as you need to reload all the same documents everytime you toggle off. 

In the React state, try to keep the original documents. So your state `documents` is only set once when you load the page and shouldn't change anymore after. Now, in the `handleToggle`, you just need to set the `toggled` state as you did and nothing else, don't do the documents filter in handleToggle. Instead, add a new function, called for example `getFilteredDocuments`. This function needs to use the state `documents` and the state `toggled` to filter the documents and return the list of documents from the function (don't set any more state here). Finally, in the render function, you currently call the `.map()` function on the documents state, instead, call the `.map()` function on the result of the call to `getFilteredDocuments` function. And that should do it :) The benefits: you keep all the original documents in the state, which act as the source of truth for the dashboard. Filtering is done dynamically based on the value of the `toggled` state. Let me know if you have any questions.

Reply:
done